src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/rope_fusion.cpp

```diff
@@ -460,22 +463,30 @@ ov::intel_cpu::RoPEFusionGPTJ::RoPEFusionGPTJ() {
         auto new_node = std::make_shared<RoPENode>(new_args, config);
         new_node->set_friendly_name(old_node->get_friendly_name());
         ov::copy_runtime_info({pattern_map.at(varsplit).get_node_shared_ptr(),
-                                pattern_map.at(unsqueeze_sin).get_node_shared_ptr(),
-                                pattern_map.at(unsqueeze_cos).get_node_shared_ptr(),
                                 pattern_map.at(repeat_interleave_sin).get_node_shared_ptr(),
                                 pattern_map.at(repeat_interleave_cos).get_node_shared_ptr(),
                                 pattern_map.at(neg_Multiply_1177).get_node_shared_ptr(),
                                 pattern_map.at(Unsqueeze_65524).get_node_shared_ptr(),
                                 pattern_map.at(Unsqueeze_65525).get_node_shared_ptr(),
                                 pattern_map.at(stack_1182).get_node_shared_ptr(),
-                                pattern_map.at(flatten_Concat_1197).get_node_shared_ptr(),
                                 pattern_map.at(mul_cos).get_node_shared_ptr(),
                                 pattern_map.at(mul_sin).get_node_shared_ptr(),
                                 pattern_map.at(rotary_emb).get_node_shared_ptr(),
                                 pattern_map.at(cat_Concat_1211).get_node_shared_ptr(),
                                 pattern_map.at(permute_Transpose_1213).get_node_shared_ptr()},
                                 new_node);
         ov::replace_node(old_node, new_node);
+        // shapeof may be moved up from transpose to add,
+        // After RoPE fusion, shapeof must be moved to the data input of RoPE otherwise extra subgraph exists
+        std::shared_ptr<ov::Node> rotary_emb_node = pattern_map.at(rotary_emb).get_node_shared_ptr();
+        auto rotary_emb_out = rotary_emb_node->output(0);
+        if (rotary_emb_out.get_target_inputs().size() == 2) {
+            for (auto& input : rotary_emb_out.get_target_inputs()) {
+                if (ov::is_type<opset1::ShapeOf>(input.get_node())) {
+                    input.replace_source_output(pattern_map.at(view_Reshape));
+                }
+            }
+        }
```

Comment:
Just for the record, this is tricky fix for a common issue: **any unexpected branch out from the middle of a fusion pattern subgraph will create unexpected leftovers and introduces performance regressions**.

This is a generic issue for pattern-matching, and it becomes serious issue when the pattern becomes more & more complex.